### ISSUE TITLE
False ambiguity with Union

### ISSUE THREAD

**Issue by timholy** [JuliaLang/julia#20246]
```julia
julia> module A
       foo{T<:Union{Int,Int64}}(x::AbstractUnitRange{T}) = 1
       foo{T<:Union{UInt,UInt64}}(x::AbstractUnitRange{T}) = 2
       end
A

julia> Base.Test.detect_ambiguities(A)
1-element Array{Tuple{Method,Method},1}:
 (foo{T<:Int64}(x::AbstractUnitRange{T}) in A at REPL[10]:2,foo{T<:UInt64}(x::AbstractUnitRange{T}) in A at REPL[10]:3)
```

**Comment by timholy** [comment]
xref #19998

**Comment by yuyichao** [comment]
I think this is why we want a version of ambiguity detection that ignores `Union{}`?

**Comment by Keno** [comment]
Yes

**Comment by timholy** [comment]
Assuming the goal is to restore the ambiguity tests before 0.6 is out, it's going to be necessary to fix this, so I added a milestone label.

**Comment by timholy** [comment]
Hmm, it looks like this was fixed in e1010000, though it's not the default. Presumably we should either make it the default or change https://github.com/JuliaLang/julia/blob/0e970f05fc9fb16b46d285f8b3fc5ff88fe53b44/test/ambiguous.jl#L137 to include `allow_bottom=false`?
